From 65f2c21ac15fac98ab3b50dd0f1496629556e1a6 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 16 Apr 2013 12:34:42 +0900
Subject: [PATCH 586/611] ARM: shmobile: r8a7778: add ICB clock support

The clock is registered with CLKDEV_CON_ID() since there is no driver
for the ICB.
---
 arch/arm/mach-shmobile/clock-r8a7778.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-shmobile/clock-r8a7778.c b/arch/arm/mach-shmobile/clock-r8a7778.c
index 4729800..4ea19ce 100644
--- a/arch/arm/mach-shmobile/clock-r8a7778.c
+++ b/arch/arm/mach-shmobile/clock-r8a7778.c
@@ -115,6 +115,7 @@ static struct clk *main_clks[] = {
 };
 
 enum {
+	MSTP410,
 	MSTP331,
 	MSTP323, MSTP322, MSTP321,
 	MSTP311, MSTP310, MSTP309, MSTP308, MSTP307,
@@ -126,6 +127,7 @@ enum {
 	MSTP_NR };
 
 static struct clk mstp_clks[MSTP_NR] = {
+	[MSTP410] = SH_CLK_MSTP32(&s3_clk, MSTPCR4, 10, 0), /* ICB */
 	[MSTP331] = SH_CLK_MSTP32(&s4_clk, MSTPCR3, 31, 0), /* MMC */
 	[MSTP323] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 23, 0), /* SDHI0 */
 	[MSTP322] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 22, 0), /* SDHI1 */
@@ -171,6 +173,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_CON_ID("audio_clk_internal",	&s1_clk),
 
 	/* MSTP32 clocks */
+	CLKDEV_CON_ID("icb",		&mstp_clks[MSTP410]), /* ICB */
 	CLKDEV_DEV_ID("sh_mmcif", &mstp_clks[MSTP331]), /* MMC */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.0", &mstp_clks[MSTP323]), /* SDHI0 */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.1", &mstp_clks[MSTP322]), /* SDHI1 */
-- 
1.8.2

